bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/util/HexUtils.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.util;
+
+/**
+ * Static utility methods that are helpful when dealing with hex data and byte arrays.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ *
+ */
+public class HexUtils {
+
+    // used for hex conversions
+    private final static char[] hexArray = "0123456789ABCDEF".toCharArray();
+
+    // private constructor as we only have static methods
+    private HexUtils() {
+    }
+
+    /**
+     * Converts a byte array into a hex string (in format "01 23 45 67 89 AB CD EF").
+     *
+     * @param bytes the byte array
+     * @return the corresponding hex string
+     */
+    public static String bytesToHexWithSpaces(byte[] bytes) {
+        char[] hexChars = new char[bytes.length * 3];
+        for (int j = 0; j < bytes.length; j++) {
+            int v = bytes[j] & 0xFF;
+            hexChars[j * 3] = hexArray[v >>> 4];
+            hexChars[j * 3 + 1] = hexArray[v & 0x0F];
+            hexChars[j * 3 + 2] = ' ';
```

Comment:
I actually see use-cases for other delimiters - e.g. `:` for MAC addresses. I'd therefore very much prefer to have the delimiter as an (optional) argument. I care less about _one_ implementation that we hopefully will never have to touch again than having an awkward API with several similar methods or, even worse, people starting to `result = result.replaceAll(" ", ":")`. And while thinking about it, another argument for a prefix would make sense, so the result can be prefixed with e.g. `0x` or `#` optionally.

Reply:
Do you have a suggestion for the method signature if you want to have prefixes and suffixes at any length? Pass two nullable Strings as additional parameters? Define a formatting (which syntax) for the single values?

Reply:
How about 

```java
public static String bytesToHex(byte[] bytes, @Nullable CharSequence prefix, @Nullable CharSequence delimiter) {
  // ...
}
```

? Of course, a convenience signature like 

```java
public static String bytesToHex(byte[] bytes) {
  return bytesToHex(bytes, null, null);
}
```

makes sense too.

Reply:
What should the convenience method return? The string with space delimiters as right now?